pkg/server/server.go

```diff
@@ -96,19 +96,29 @@ func renderer(templatePath string) multitemplate.Renderer {
 	return r
 }
 
-func requireAuth(db *sqlx.DB, secret string) func(*gin.Context) {
+func requireTokenAuth(db *sqlx.DB, secret, authType string) func(*gin.Context) {
 	return func(ctx *gin.Context) {
-		jobID := ctx.Param("id")
-		job, err := data.GetJob(jobID, db)
-		if err != nil {
-			log.Println(fmt.Errorf("requireAuth failed to getJob: %w", err))
-			ctx.AbortWithStatus(http.StatusInternalServerError)
-			return
-		}
+    var expected string
+
+    switch authType {
+    case "job":
+      jobID := ctx.Param("id")
+      job, err := data.GetJob(jobID, db)
+      if err != nil || job.ID != jobID {
```

Comment:
In most cases for security I would say we should just return 500 for both checks but in this case I think it might be best to return 404 for `job.ID != jobID`.